behave/features/steps/frontend_complaints.py

```diff
@@ -0,0 +1,49 @@
+from behave import step
+from common_steps import click_on_hyperlink_and_get_href, assert_header_on_page
+
+
+@step("I am on the '{tab}' tab on the dashboard")
+def step_impl_complaints_tab(context, tab):
+    click_on_hyperlink_and_get_href(context, tab)
+
+
+@step("There are complaints available")
+def step_impl_complaints_cases(context):
+    assert (
+        context.helperfunc.find_by_xpath(
+            "//table[contains(@class,'ListTable ng-scope')]/tbody/tr"
+        )
+        is not None
+    ), "No Complaint Cases Available"
+
+
+@step("I search for '{complaint_text}'")
+def step_impl_complaints_search(context, complaint_text):
+    context.helperfunc.find_by_id("case-search").send_keys(complaint_text)
```

Comment:
Consider using clear() before sending keys. Although we know in this instance that the text field would be empty. In an instance where a text field was not empty, we would be appending it with whatever we are sending.